src/commands/dailyCheckIn/scheduleCheckIn.js

```diff
@@ -0,0 +1,186 @@
+/**
+ * Author: Arthur Powers
+ * Date: 5/22/24
+ * 
+ * Initiate a prompt allowing users to create a scheduler.
+ */
+
+const { PermissionFlagsBits, ApplicationCommandOptionType, Client, CommandInteraction } = require("discord.js");
+const dayjs = require('dayjs');
+const utc = require('dayjs/plugin/utc');
+const weekday = require('dayjs/plugin/weekday');
+dayjs.extend(utc);
+dayjs.extend(weekday);
+
+
+const fakeScheduleEntry = {
+    "monday": {},
+    "tuesday": {},
+    "wednesday": {},
+    "thursday": {},
+    "friday": {},
+    "saturday": {},
+    "sunday": {},
+}
+
+module.exports = {
+    /**
+     * Parse and save schedule date to the database
+     * @param {Client} client 
+     * @param {CommandInteraction} interaction 
+     */
+    callback: async (client, interaction) => {
+
+        const userId = interaction?.user?.id;
+        const userTag = interaction?.user?.tag;
+
+        await interaction.deferReply({ ephemeral: true });
+
+        try {
+
+            // command should include a user
+            if (userId === undefined || userTag === undefined) {
+                await interaction.editReply({
+                    ephemeral: true,
+                    content: 'Could not process command'
+                });
+                return;
+            }
+
+
+            const rawDays = interaction.options.get('days')?.value;
+            const rawTime = interaction.options.get('time')?.value;
+            let timeHours;
+            let timeMinutes;
+            let parsedDays = [];
+            const validDays = ['sunday', 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday'];
+            const daily = rawDays.toLocaleLowerCase().startsWith('daily');
+
+            // split days by whitespace or comma
+            if (daily) {
+                parsedDays = validDays;
+            }
+            else {
+                parsedDays = rawDays.split(/[\s,]+/i).map(s => s.toString().toLocaleLowerCase());
+            }
+
+            // check if parsed days are all valid
+            if (parsedDays.some(d => !validDays.includes(d))) {
+                await interaction.editReply({
+                    ephemeral: true,
+                    content: 'Invalid list of days'
+                });
+                return;
+            }
+
+            // regex checks for [number]:[number]p
+            // am/pm is determined if 'p' exists
+            // regex returns array of separated parts [original, hour, minute, pm]
+            var parsedTime = rawTime.match(/(\d+)(?::(\d\d))?\s*(p?)/);
+
+            try {
```

Comment:
Yeah I think better to use Day.js. See this part especially: `If you don't know the exact format of an input string, but know it could be one of many, you can use an array of formats.` https://day.js.org/docs/en/parse/string-format